### PR TITLE
Simplify customer form

### PR DESCRIPTION
**Pull request by JevgenijVisockij**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies import form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Simplifying Sell -> Customer edit and view forms. Everything must look the same with one exception: blue help box replaced with help text under input.

:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType by CustomerType, TransferGuestAccountType. This means if any module extends any of this types they will get an exception upon upgrading to PS version containing changes in this PR.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21243)

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 1]
```diff
@@ -113,89 +114,144 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
             ->add('first_name', TextType::class, [
+                'label' => $this->trans('First name', 'Admin.Global'),
+                'help' => $this->trans(
+                    'Only letters and the dot (.) character, followed by a space, are allowed.',
+                    'Admin.Orderscustomers.Help'
+                ),
                 'constraints' => [
                     new NotBlank([
-                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
+                        'message' => $this->trans('This field cannot be empty', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                        'message' => $this->trans('This field cannot be empty.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 2]
```diff
@@ -113,89 +114,144 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
             ->add('last_name', TextType::class, [
+                'label' => $this->trans('Last name', 'Admin.Global'),
+                'help' => $this->trans(
+                    'Only letters and the dot (.) character, followed by a space, are allowed.',
+                    'Admin.Orderscustomers.Help'
+                ),
                 'constraints' => [
                     new NotBlank([
-                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
+                        'message' => $this->trans('This field cannot be empty', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                        'message' => $this->trans('This field cannot be empty.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 3]
```diff
@@ -113,89 +114,144 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                        ),
                     ]),
                 ],
             ])
             ->add('email', EmailType::class, [
+                'label' => $this->trans('Email', 'Admin.Global'),
                 'constraints' => [
                     new NotBlank([
-                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
+                        'message' => $this->trans('This field cannot be empty', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                        'message' => $this->trans('This field cannot be empty.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 4]
```diff
@@ -113,89 +114,144 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
                     ]),
                     new Email([
-                        'message' => $this->trans('This field is invalid', [], 'Admin.Notifications.Error'),
+                        'message' => $this->trans('This field is invalid', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                        'message' => $this->trans('This field is invalid.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 5]
```diff
@@ -205,33 +261,54 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
                     'required' => false,
                 ])
                 ->add('ape_code', TextType::class, [
+                    'label' => $this->trans('APE', 'Admin.Orderscustomers.Feature'),
                     'required' => false,
                     'constraints' => [
                         new Type([
                             'type' => 'alnum',
-                            'message' => $this->trans('This field is invalid', [], 'Admin.Notifications.Error'),
+                            'message' => $this->trans('This field is invalid', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                            'message' => $this->trans('This field is invalid.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 6]
```diff
@@ -205,33 +261,54 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                        '%s 0-9',
+                        $this->trans(
+                            'Valid characters:',
+                            'Admin.Orderscustomers.Help'
+                        )
+                    ),
                     'scale' => 6,
                     'required' => false,
-                    'invalid_message' => $this->trans('This field is invalid', [], 'Admin.Notifications.Error'),
+                    'invalid_message' => $this->trans('This field is invalid', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                    'invalid_message' => $this->trans('This field is invalid.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 7]
```diff
@@ -205,33 +261,54 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                    'help' => sprintf(
+                        '%s 0-9',
+                        $this->trans(
+                            'Valid characters:',
+                            'Admin.Orderscustomers.Help'
+                        )
+                    ),
                     'required' => false,
-                    'invalid_message' => $this->trans('This field is invalid', [], 'Admin.Notifications.Error'),
+                    'invalid_message' => $this->trans('This field is invalid', 'Admin.Notifications.Error'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                    'invalid_message' => $this->trans('This field is invalid.', 'Admin.Notifications.Error'),
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/Blocks/Index/required_fields.html.twig [hunk 8]
```diff
@@ -22,28 +22,31 @@
...
-    <div class="card-body">
-      <div class="alert alert-info" role="alert">
-        <div class="alert-text">
-          <p>{{ 'Select the fields you would like to be required for this section.'|trans({}, 'Admin.Orderscustomers.Help') }}</p>
-          <p>{{ 'Please make sure you are complying with the opt-in legislation applicable in your country.'|trans({}, 'Admin.Orderscustomers.Help') }}</p>
+  {% block customer_required_fields_form %}
+    {{ form_start(customerRequiredFieldsForm, {'action': path('admin_customers_set_required_fields')}) }}
+    <div class="card" >
+      <h3 class="card-header">
+        {{ 'Required Fields'|trans({}, 'Admin.Orderscustomers.Feature') }}
```
Comment (Julievrz) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Customer/Blocks/Index/required_fields.html.twig]:
```suggestion
        {{ 'Required fields'|trans({}, 'Admin.Orderscustomers.Feature') }}
```

**Review by Julievrz — COMMENTED** [review]
"First name" should be located in Admin.Global
"Enabled" should be located in Admin.Global
"Only the discount for the selected group will be applied to this customer." should be located in Admin.Oderscustomers.Help.

If I am not mistaken, these wordings are correctly located in the code but I don't know why the bot has written wrong domains.

**Comment by JevgenijVisockij** [comment]
> "First name" should be located in Admin.Global
> "Enabled" should be located in Admin.Global
> "Only the discount for the selected group will be applied to this customer." should be located in Admin.Oderscustomers.Help.
> 
> If I am not mistaken, these wordings are correctly located in the code but I don't know why the bot has written wrong domains.

Because I fixed them after I saw bots message. I noticed that sometimes bot removes or modifies its comment automatically sometimes it doesn't.

**Comment by Julievrz** [comment]
> > "First name" should be located in Admin.Global
> > "Enabled" should be located in Admin.Global
> > "Only the discount for the selected group will be applied to this customer." should be located in Admin.Oderscustomers.Help.
> > If I am not mistaken, these wordings are correctly located in the code but I don't know why the bot has written wrong domains.
> 
> Because I fixed them after I saw bots message. I noticed that sometimes bot removes or modifies its comment automatically sometimes it doesn't.

Ok that's why! Thank you for letting me know. ☺️

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

I found these issues:

1. Go to BO > Customers > Customers lit page
2. Click on "Set required fields for this section"
3. See error

image
**Without PR**
image

4. Create a new customer
5. Scroll down to the Group access section
6. See error
image

**Without PR**
image

Thanks!

**Comment by JevgenijVisockij** [comment]
Hey!, Thanks for the review! Fixed the issue.

**Review by atomiix — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PrestaShop/prestashop-core-developers,

This PR is ready for QA?

Thanks!

**Comment by JevgenijVisockij** [comment]
> Hi @PrestaShop/prestashop-core-developers,
> 
> This PR is ready for QA?
> 
> Thanks!

Hey! From my side, yeah.

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

It is ok :heavy_check_mark: 
https://www.awesomescreenshot.com/video/1747875?key=45550947bb683bca7bda1fcb3f938777

Thanks!

**Comment by Progi1984** [comment]
Thanks @JevgenijVisockij & @khouloudbelguith

**Pull request merged**